### PR TITLE
Add RowNumberNode

### PR DESCRIPTION
**Pull request by mbasmanova**
RowNumberNode is a special case of a WindowNode with a single row_number window
function, no sorting and an optional limit. This node can be evaluated more
efficiently then generic WindowNode. 

A follow-up PR will introduce RowNumber operator.

### CODE REVIEW

File: velox/core/PlanNode.h [hunk 1]
```diff
@@ -1936,4 +1936,53 @@ class WindowNode : public PlanNode {
   const RowTypePtr outputType_;
 };
 
+/// Optimized version of a WindowNode for a single row_number function with an
+/// optional limit and no sorting.
```
Comment (aditi-pandit) [inline velox/core/PlanNode.h:1940]: It seems like even partition keys could be empty. Please can you add this info to the comment.

Comment (mbasmanova) [reply velox/core/PlanNode.h:1940]: @aditi-pandit Sure. Updated.

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks!

**Review by aditi-pandit — APPROVED** [review]
Thanks @mbasmanova

**Pull request closed without merging**